pkg/cluster/sshkeys.go

```diff
@@ -11,33 +11,24 @@ import (
 	"math/big"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database"
 )
 
-func (m *manager) generateSSHKey(ctx context.Context) error {
-	var err error
-	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-		if doc.OpenShiftCluster.Properties.SSHKey == nil {
-			sshKey, err := rsa.GenerateKey(rand.Reader, 2048)
-			if err != nil {
-				return err
-			}
+func (m *manager) ensureSSHKey(ctx context.Context) error {
+	sshKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
 
+	var f database.OpenShiftClusterDocumentMutator = func(doc *api.OpenShiftClusterDocument) error {
```

Comment:
I am not sure I like the name of this, something a little bit more expressive would be better imho
I don't really like when we declare functions like that btw, it makes it impossible to test them. Here it is not a big big deal, but it is something to keep in mind